File: floatingnotification/src/main/java/com/example/floatingnotification/utils/DimensUtils.java
package com.example.floatingnotification.utils;

import android.content.Context;
import android.content.res.Resources;
import android.util.DisplayMetrics;

public class DimensUtils {

    public static int convertDpToPixel(int dps, Context context) {
        final float scale = context.getResources().getDisplayMetrics().density;
        int pixels = (int) (dps * scale + 0.5f);
        return pixels;
    }

    public static int convertDpToPixel(float dps, Context context) {
        final float scale = context.getResources().getDisplayMetrics().density;
        int pixels = (int) (dps * scale + 0.5f);
        return pixels;
    }

    public static int convertPixelsToDp(int px, Context context) {
        Resources resources = context.getResources();
        DisplayMetrics metrics = resources.getDisplayMetrics();
        int dp = (int) (px / metrics.density);
        return dp;
    }
}

File: floatingnotification/src/main/java/com/example/floatingnotification/utils/DimensUtils.kt
package com.example.floatingnotification.utils

import android.content.Context

class DimensUtils {
    companion object {
        fun convertDpToPixel(dps: Int, context: Context): Int {
            val scale = context.resources.displayMetrics.density
            return (dps * scale + 0.5f).toInt()
        }

        fun convertDpToPixel(dps: Float, context: Context): Int {
            val scale = context.resources.displayMetrics.density
            return (dps * scale + 0.5f).toInt()
        }

        fun convertPixelsToDp(px: Int, context: Context): Int {
            val resources = context.resources
            val metrics = resources.displayMetrics
            return (px / metrics.density).toInt()
        }
    }
}